python/pyspark/pandas/frame.py

```diff
@@ -5500,6 +5500,25 @@ def op(psser: ps.Series) -> ps.Series:
         else:
             return psdf
 
+    def interpolate(
+        self, method: Optional[str] = None, limit: Optional[int] = None
+    ) -> Optional["DataFrame"]:
+        if (method is not None) and (method not in ["linear"]):
+            raise ValueError("Expecting 'linear'.")
```

Comment:
How about raising `NotImplementedError` for methods that pandas support?

Reply:
good catch! will update